diff -ur dbi-0.1.0-o/setup.rb dbi-0.1.0/setup.rb
--- dbi-0.1.0-o/setup.rb	2006-02-13 06:35:18.000000000 -0700
+++ dbi-0.1.0/setup.rb	2006-03-14 17:57:57.000000000 -0700
@@ -269,14 +269,14 @@
 						isdir( File.join( @config['rb-dir'], 'DBD', capname ) )
 						install(
 							basefile,
-							File.join( @config['rb-dir'], 'DBD', capname, basefile ),
+							File.join( @config['prefix'], @config['rb-dir'], 'DBD', capname, basefile ),
 							0644
 						)
 					end
 				end
 			}
 			into_dir( 'dbi' ) { install_rb 'dbi' }
-			install( 'dbi.rb', @config['rb-dir'], 0644 )
+			install( 'dbi.rb', File.join(@config['prefix'], @config['rb-dir']), 0644 )
     }
     into_dir( 'ext' ) {
       foreach_package do |targ, *dummy|
@@ -372,6 +372,9 @@
     'bin-dir'   => [ bindir,
                      'path',
                      'directory to install commands' ],
+    'prefix'   => [ '',
+                     'path',
+                     'directory to prefix install' ],
     'rb-dir'    => [ siterb,
                      'path',
                      'directory to install ruby scripts' ],
@@ -399,7 +402,7 @@
                      'package name(s) you do not want to install' ]
   }
 
-  CONFIG_ORDER = %w( bin-dir rb-dir so-dir ruby-path make-prog with without )
+  CONFIG_ORDER = %w( bin-dir rb-dir so-dir ruby-path make-prog with without prefix )
 
   def save_configs
     File.open( ConfigFile, 'w' ) do |f|
@@ -573,7 +576,7 @@
       next if /\A\./ === fname
       next unless File.file? fname
 
-      install fname, dir, mode
+      install fname, File.join(@config['prefix'], dir), mode
     end
   end
 
@@ -601,7 +604,7 @@
   def install_so( dir )
     to = isdir(File.expand_path(@config['so-dir'] + '/' + dir))
     find_so('.').each do |fn|
-      install fn, to, 0555
+      install fn, File.join(@config['prefix'], to), 0555
     end
   end
 
@@ -658,7 +661,7 @@
       dirs[0] = disk + dirs[0]
     end
     dirs.each_index do |idx|
-      path = dirs[0..idx].join('')
+      path = File.join(@config['prefix'], dirs[0..idx].join(''))
       Dir.mkdir path unless dir? path
     end
   end
Only in dbi-0.1.0: setup.rb.orig
